main.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+
+	"cloud.google.com/go/logging/jsonlog"
+	"example.com/micro/metadata"
+	"github.com/gorilla/mux"
+)
+
+type App struct {
+	*http.Server
+	projectID string
+	log       *jsonlog.Logger
+}
+
+func main() {
+	ctx := context.Background()
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+	}
+	log.Printf("listening on port %s", port)
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	app, err := newApp(ctx, port, projectID)
+	if err != nil {
+		log.Fatalf("unable to initialize application: %v", err)
+	}
+	log.Println("starting HTTP server")
+	go func() {
+		if err := app.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			log.Fatalf("Server closed: %v", err)
+		}
+	}()
+
+	// Listen for SIGINT to gracefully shutdown.
```

Comment:
This is nice for local dev, but for Cloud Run  it needs to be SIGTERM

Reply:
Looking for both now.